// @hardpool/ngx-spinner/esm5/src/app/ngx-spinner/blink/blink.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
var BlinkComponent = /** @class */ (function () {
    function BlinkComponent() {
    }
    /**
     * @return {?}
     */
    BlinkComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    BlinkComponent.decorators = [
        { type: Component, args: [{
                    selector: 'spinner-blink',
                    template: "<div class=\"sk-scaleout\" [style.height]=\"config.size\" [style.width]=\"config.size\" [style.background]=\"config.color\">\n</div>",
                    styles: [".sk-scaleout{margin:0;border-radius:100%;-webkit-animation:1s ease-in-out infinite sk-scaleout;animation:1s ease-in-out infinite sk-scaleout}@-webkit-keyframes sk-scaleout{0%{-webkit-transform:scale(0)}100%{-webkit-transform:scale(1);opacity:0}}@keyframes sk-scaleout{0%{-webkit-transform:scale(0);transform:scale(0)}100%{-webkit-transform:scale(1);transform:scale(1);opacity:0}}"]
                },] },
    ];
    /** @nocollapse */
    BlinkComponent.ctorParameters = function () { return []; };
    return BlinkComponent;
}());
export { BlinkComponent };
function BlinkComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    BlinkComponent.prototype.config;
}
