// @hardpool/ngx-spinner/esm5/src/app/ngx-spinner/plate-rotate/plate-rotate.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
var PlateRotateComponent = /** @class */ (function () {
    function PlateRotateComponent() {
    }
    /**
     * @return {?}
     */
    PlateRotateComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    PlateRotateComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-plate-rotate',
                    template: "<div class=\"spinner\" [style.height]=\"config.size\" [style.width]=\"config.size\" [style.background]=\"config.color\"\t></div>",
                    styles: [".spinner{width:40px;height:40px;-webkit-animation:1.2s ease-in-out infinite sk-rotateplane;animation:1.2s ease-in-out infinite sk-rotateplane}@-webkit-keyframes sk-rotateplane{0%{-webkit-transform:perspective(120px)}50%{-webkit-transform:perspective(120px) rotateY(180deg)}100%{-webkit-transform:perspective(120px) rotateY(180deg) rotateX(180deg)}}@keyframes sk-rotateplane{0%{transform:perspective(120px) rotateX(0) rotateY(0);-webkit-transform:perspective(120px) rotateX(0) rotateY(0)}50%{transform:perspective(120px) rotateX(-180.1deg) rotateY(0);-webkit-transform:perspective(120px) rotateX(-180.1deg) rotateY(0)}100%{transform:perspective(120px) rotateX(-180deg) rotateY(-179.9deg);-webkit-transform:perspective(120px) rotateX(-180deg) rotateY(-179.9deg)}}"]
                },] },
    ];
    /** @nocollapse */
    PlateRotateComponent.ctorParameters = function () { return []; };
    return PlateRotateComponent;
}());
export { PlateRotateComponent };
function PlateRotateComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    PlateRotateComponent.prototype.config;
}
