// @hardpool/ngx-spinner/bundles/hardpool-ngx-spinner.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('@hardpool/ngx-spinner', ['exports', '@angular/core', '@angular/common'], factory) :
    (factory((global.hardpool = global.hardpool || {}, global.hardpool['ngx-spinner'] = {}),global.ng.core,global.ng.common));
}(this, (function (exports,core,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxSpinnerService = (function () {
        function NgxSpinnerService() {
        }
        NgxSpinnerService.globalSpinnerConf = {};
        NgxSpinnerService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        NgxSpinnerService.ctorParameters = function () { return []; };
        return NgxSpinnerService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /** @enum {string} */
    var SPINNER_PLACEMENT = {
        block_window: "block_window",
        block_ui: "block_ui",
        inplace: "inplace",
    };
    /** @enum {number} */
    var SPINNER_ANIMATIONS = {
        blink: 0,
        fb_card: 1,
        plate_rotate: 2,
        double_bounce: 3,
        folding_cube: 4,
        bars: 5,
        rotating_dots: 6,
        horizontal_bounce: 7,
        spin_1: 8,
        spin_2: 9,
        spin_3: 10,
    };
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.blink] = "blink";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.fb_card] = "fb_card";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.plate_rotate] = "plate_rotate";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.double_bounce] = "double_bounce";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.folding_cube] = "folding_cube";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.bars] = "bars";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.rotating_dots] = "rotating_dots";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.horizontal_bounce] = "horizontal_bounce";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_1] = "spin_1";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_2] = "spin_2";
    SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_3] = "spin_3";
    var /** @type {?} */ DEFAULT_CONFIG = {
        placement: SPINNER_PLACEMENT.inplace,
        animation: SPINNER_ANIMATIONS.blink,
        size: "2rem",
        bgColor: "rgba(255,255,255,0.8)",
        color: "#e46521",
        secondaryColor: "#1574b3",
        tertiaryColor: "#c31a4f"
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxSpinner = (function () {
        function NgxSpinner(ngxSpinnerService) {
            this.ngxSpinnerService = ngxSpinnerService;
            this._config = {};
            // assign default config
            this.assignConfig(DEFAULT_CONFIG);
            // override wtih globle config
            this.assignConfig(NgxSpinnerService.globalSpinnerConf);
            // hide spinner as default
            this._isVisible = false;
        }
        Object.defineProperty(NgxSpinner.prototype, "visible", {
            get: /**
             * @return {?}
             */ function () {
                return this._isVisible;
            },
            set: /**
             * @param {?} condition
             * @return {?}
             */ function (condition) {
                if (condition)
                    this.show();
                else
                    this.hide();
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(NgxSpinner.prototype, "config", {
            get: /**
             * @return {?}
             */ function () {
                return this._config;
            },
            set: /**
             * @param {?} conf
             * @return {?}
             */ function (conf) {
                // override with spinner config
                this.assignConfig(conf);
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        NgxSpinner.prototype.show = /**
         * @return {?}
         */
            function () {
                this._isVisible = true;
            };
        /**
         * @return {?}
         */
        NgxSpinner.prototype.hide = /**
         * @return {?}
         */
            function () {
                this._isVisible = false;
            };
        /**
         * @return {?}
         */
        NgxSpinner.prototype.toggle = /**
         * @return {?}
         */
            function () {
                this._isVisible = !this._isVisible;
            };
        /**
         * @param {?} conf
         * @return {?}
         */
        NgxSpinner.prototype.assignConfig = /**
         * @param {?} conf
         * @return {?}
         */
            function (conf) {
                var /** @type {?} */ cnf = Object.assign(conf);
                for (var /** @type {?} */ prop in cnf) {
                    if (cnf.hasOwnProperty(prop))
                        this._config[prop] = cnf[prop];
                }
            };
        NgxSpinner.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-spinner',
                        template: "<!-- spinner -->\n<ng-template *ngIf=\"visible\" [spinner-animation]=\"config\"></ng-template>",
                        styles: [":host.block_window{align-items:center;justify-content:center;position:fixed;display:flex;width:100%;height:100%;top:0;left:0;z-index:10002}:host.block_ui{align-items:center;justify-content:center;position:absolute;display:flex;width:100%;height:100%;top:0;left:0;z-index:10001}:host.inplace{position:absolute;background:0 0!important;display:flex;width:100%;height:100%;top:0;left:0;align-items:center;justify-content:start}"],
                        host: { '[class]': 'visible?config.placement:""', '[style.background]': 'visible?config.bgColor:""' }
                    },] },
        ];
        /** @nocollapse */
        NgxSpinner.ctorParameters = function () {
            return [
                { type: NgxSpinnerService }
            ];
        };
        NgxSpinner.propDecorators = {
            visible: [{ type: core.Input }],
            config: [{ type: core.Input }]
        };
        return NgxSpinner;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var BlinkComponent = (function () {
        function BlinkComponent() {
        }
        /**
         * @return {?}
         */
        BlinkComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        BlinkComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'spinner-blink',
                        template: "<div class=\"sk-scaleout\" [style.height]=\"config.size\" [style.width]=\"config.size\" [style.background]=\"config.color\">\n</div>",
                        styles: [".sk-scaleout{margin:0;border-radius:100%;-webkit-animation:1s ease-in-out infinite sk-scaleout;animation:1s ease-in-out infinite sk-scaleout}@-webkit-keyframes sk-scaleout{0%{-webkit-transform:scale(0)}100%{-webkit-transform:scale(1);opacity:0}}@keyframes sk-scaleout{0%{-webkit-transform:scale(0);transform:scale(0)}100%{-webkit-transform:scale(1);transform:scale(1);opacity:0}}"]
                    },] },
        ];
        /** @nocollapse */
        BlinkComponent.ctorParameters = function () { return []; };
        return BlinkComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var FbCardComponent = (function () {
        function FbCardComponent() {
        }
        /**
         * @return {?}
         */
        FbCardComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        FbCardComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-fb-card',
                        template: "<div class=\"FB-Loading-Card\" [style.height]=\"config.size\" [style.width]=\"config.size\">\n\t<div>\n\t <div [style.background]=\"config.color\"></div>\n\t <div [style.background]=\"config.color\"></div>\n\t <div [style.background]=\"config.color\"></div>\n\t</div>\n\t<div [style.background]=\"config.color\"></div>\n\t<div [style.background]=\"config.color\"></div>\n\t<div [style.background]=\"config.color\"></div>\n </div>",
                        styles: ["@charset \"UTF-8\";.FB-Loading-Card{display:block;padding:12px}.FB-Loading-Card>div:nth-child(1){width:100%;height:30px}.FB-Loading-Card>div>div:nth-child(1){position:relative;left:0;height:40px;width:40px;-webkit-animation-name:shine;animation-name:shine;-webkit-animation-duration:2s;animation-duration:2s;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite;-webkit-animation-timing-function:linear;animation-timing-function:linear}.FB-Loading-Card>div:nth-child(2),.FB-Loading-Card>div:nth-child(3),.FB-Loading-Card>div:nth-child(4),.FB-Loading-Card>div>div:nth-child(2),.FB-Loading-Card>div>div:nth-child(3){position:relative;height:6px;-webkit-animation-name:shine;animation-name:shine;-webkit-animation-duration:2s;animation-duration:2s;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite;-webkit-animation-timing-function:linear;animation-timing-function:linear}@-webkit-keyframes shine{from{transition:none}to{background-color:#f6f7f8;transition:all .3s ease-out}}@keyframes shine{from{transition:none}to{background-color:#f6f7f8;transition:all .3s ease-out}}.FB-Loading-Card>div>div:nth-child(2){left:50px;top:-33px;width:25%}.FB-Loading-Card>div>div:nth-child(3){left:50px;top:-20px;width:15%}.FB-Loading-Card>div:nth-child(2){left:0;top:30px;width:80%}.FB-Loading-Card>div:nth-child(3){left:0;top:45px;width:90%}.FB-Loading-Card>div:nth-child(4){left:0;top:60px;width:50%}"]
                    },] },
        ];
        /** @nocollapse */
        FbCardComponent.ctorParameters = function () { return []; };
        return FbCardComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var PlateRotateComponent = (function () {
        function PlateRotateComponent() {
        }
        /**
         * @return {?}
         */
        PlateRotateComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        PlateRotateComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-plate-rotate',
                        template: "<div class=\"spinner\" [style.height]=\"config.size\" [style.width]=\"config.size\" [style.background]=\"config.color\"\t></div>",
                        styles: [".spinner{width:40px;height:40px;-webkit-animation:1.2s ease-in-out infinite sk-rotateplane;animation:1.2s ease-in-out infinite sk-rotateplane}@-webkit-keyframes sk-rotateplane{0%{-webkit-transform:perspective(120px)}50%{-webkit-transform:perspective(120px) rotateY(180deg)}100%{-webkit-transform:perspective(120px) rotateY(180deg) rotateX(180deg)}}@keyframes sk-rotateplane{0%{transform:perspective(120px) rotateX(0) rotateY(0);-webkit-transform:perspective(120px) rotateX(0) rotateY(0)}50%{transform:perspective(120px) rotateX(-180.1deg) rotateY(0);-webkit-transform:perspective(120px) rotateX(-180.1deg) rotateY(0)}100%{transform:perspective(120px) rotateX(-180deg) rotateY(-179.9deg);-webkit-transform:perspective(120px) rotateX(-180deg) rotateY(-179.9deg)}}"]
                    },] },
        ];
        /** @nocollapse */
        PlateRotateComponent.ctorParameters = function () { return []; };
        return PlateRotateComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var DoubleBounceComponent = (function () {
        function DoubleBounceComponent() {
        }
        /**
         * @return {?}
         */
        DoubleBounceComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        DoubleBounceComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-double-bounce',
                        template: "<div class=\"spinner\" [style.height]=\"config.size\" [style.width]=\"config.size\">\n  <div class=\"double-bounce1\" [style.background]=\"config.color\"></div>\n  <div class=\"double-bounce2\" [style.background]=\"config.secondaryColor\"></div>\n</div>",
                        styles: [".spinner{position:relative}.double-bounce1,.double-bounce2{width:100%;height:100%;border-radius:50%;opacity:.6;position:absolute;top:0;left:0;-webkit-animation:2s ease-in-out infinite sk-bounce;animation:2s ease-in-out infinite sk-bounce}.double-bounce2{-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-bounce{0%,100%{-webkit-transform:scale(0)}50%{-webkit-transform:scale(1)}}@keyframes sk-bounce{0%,100%{transform:scale(0);-webkit-transform:scale(0)}50%{transform:scale(1);-webkit-transform:scale(1)}}"]
                    },] },
        ];
        /** @nocollapse */
        DoubleBounceComponent.ctorParameters = function () { return []; };
        return DoubleBounceComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var FoldingCubeComponent = (function () {
        function FoldingCubeComponent() {
        }
        /**
         * @return {?}
         */
        FoldingCubeComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.styles.nativeElement.innerHTML = "<style>.sk-folding-cube .sk-cube:before { background-color: " + this.config.color + " }</style>";
            };
        FoldingCubeComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-folding-cube',
                        template: "<div class=\"sk-folding-cube\" [style.height]=\"config.size\" [style.width]=\"config.size\">\n  <div class=\"sk-cube1 sk-cube\"></div>\n  <div class=\"sk-cube2 sk-cube\"></div>\n  <div class=\"sk-cube4 sk-cube\"></div>\n  <div class=\"sk-cube3 sk-cube\"></div>\n</div>\n<div #styles></div>",
                        styles: [".sk-folding-cube{position:relative;-webkit-transform:rotateZ(45deg);transform:rotateZ(45deg)}.sk-folding-cube .sk-cube{float:left;width:50%;height:50%;position:relative;-webkit-transform:scale(1.1);transform:scale(1.1)}.sk-folding-cube .sk-cube:before{content:'';position:absolute;top:0;left:0;width:100%;height:100%;-webkit-animation:2.4s linear infinite both sk-foldCubeAngle;animation:2.4s linear infinite both sk-foldCubeAngle;-webkit-transform-origin:100% 100%;transform-origin:100% 100%}.sk-folding-cube .sk-cube2{-webkit-transform:scale(1.1) rotateZ(90deg);transform:scale(1.1) rotateZ(90deg)}.sk-folding-cube .sk-cube3{-webkit-transform:scale(1.1) rotateZ(180deg);transform:scale(1.1) rotateZ(180deg)}.sk-folding-cube .sk-cube4{-webkit-transform:scale(1.1) rotateZ(270deg);transform:scale(1.1) rotateZ(270deg)}.sk-folding-cube .sk-cube2:before{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-folding-cube .sk-cube3:before{-webkit-animation-delay:.6s;animation-delay:.6s}.sk-folding-cube .sk-cube4:before{-webkit-animation-delay:.9s;animation-delay:.9s}@-webkit-keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}@keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}"]
                    },] },
        ];
        /** @nocollapse */
        FoldingCubeComponent.ctorParameters = function () { return []; };
        FoldingCubeComponent.propDecorators = {
            styles: [{ type: core.ViewChild, args: ["styles",] }]
        };
        return FoldingCubeComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var BarsComponent = (function () {
        function BarsComponent() {
        }
        /**
         * @return {?}
         */
        BarsComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        BarsComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-bars',
                        template: "<div class=\"spinner\" [style.height]=\"config.size\" >\n  <div class=\"rect1\" [style.background]=\"config.color\"></div>\n  <div class=\"rect2\" [style.background]=\"config.color\"></div>\n  <div class=\"rect3\" [style.background]=\"config.color\"></div>\n  <div class=\"rect4\" [style.background]=\"config.color\"></div>\n  <div class=\"rect5\" [style.background]=\"config.color\"></div>\n</div>",
                        styles: [".spinner{margin:0}.spinner>div{height:100%;width:.4rem;margin:0 .15rem;display:inline-block;-webkit-animation:1.2s ease-in-out infinite sk-stretchdelay;animation:1.2s ease-in-out infinite sk-stretchdelay}.spinner .rect2{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.spinner .rect3{-webkit-animation-delay:-1s;animation-delay:-1s}.spinner .rect4{-webkit-animation-delay:-.9s;animation-delay:-.9s}.spinner .rect5{-webkit-animation-delay:-.8s;animation-delay:-.8s}@-webkit-keyframes sk-stretchdelay{0%,100%,40%{-webkit-transform:scaleY(.4)}20%{-webkit-transform:scaleY(1)}}@keyframes sk-stretchdelay{0%,100%,40%{transform:scaleY(.4);-webkit-transform:scaleY(.4)}20%{transform:scaleY(1);-webkit-transform:scaleY(1)}}"]
                    },] },
        ];
        /** @nocollapse */
        BarsComponent.ctorParameters = function () { return []; };
        return BarsComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var RotatingDotsComponent = (function () {
        function RotatingDotsComponent() {
        }
        /**
         * @return {?}
         */
        RotatingDotsComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        RotatingDotsComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-rotating-dots',
                        template: "<div class=\"spinner\" [style.height]=\"config.size\" [style.width]=\"config.size\">\n  <div class=\"dot1\" [style.background]=\"config.color\"></div>\n  <div class=\"dot2\" [style.background]=\"config.color\"></div>\n</div>",
                        styles: [".spinner{position:relative;text-align:center;-webkit-animation:2s linear infinite sk-rotate;animation:2s linear infinite sk-rotate}.dot1,.dot2{width:60%;height:60%;display:inline-block;position:absolute;top:0;border-radius:100%;-webkit-animation:2s ease-in-out infinite sk-bounce;animation:2s ease-in-out infinite sk-bounce}.dot2{top:auto;bottom:0;-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-rotate{100%{-webkit-transform:rotate(360deg)}}@keyframes sk-rotate{100%{transform:rotate(360deg);-webkit-transform:rotate(360deg)}}@-webkit-keyframes sk-bounce{0%,100%{-webkit-transform:scale(0)}50%{-webkit-transform:scale(1)}}@keyframes sk-bounce{0%,100%{transform:scale(0);-webkit-transform:scale(0)}50%{transform:scale(1);-webkit-transform:scale(1)}}"]
                    },] },
        ];
        /** @nocollapse */
        RotatingDotsComponent.ctorParameters = function () { return []; };
        return RotatingDotsComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var HorizontalBounceComponent = (function () {
        function HorizontalBounceComponent() {
        }
        /**
         * @return {?}
         */
        HorizontalBounceComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        HorizontalBounceComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-horizontal-bounce',
                        template: "<div class=\"spinner\">\n  <div class=\"bounce1\" [style.height]=\"config.size\" [style.width]=\"config.size\" [style.background]=\"config.color\"></div>\n  <div class=\"bounce2\" [style.height]=\"config.size\" [style.width]=\"config.size\" [style.background]=\"config.color\"></div>\n  <div class=\"bounce3\" [style.height]=\"config.size\" [style.width]=\"config.size\" [style.background]=\"config.color\"></div>\n</div>",
                        styles: [".spinner{text-align:center}.spinner>div{margin:0 .15rem;border-radius:100%;display:inline-block;-webkit-animation:1.4s ease-in-out infinite both sk-bouncedelay;animation:1.4s ease-in-out infinite both sk-bouncedelay}.spinner .bounce1{-webkit-animation-delay:-.32s;animation-delay:-.32s}.spinner .bounce2{-webkit-animation-delay:-.16s;animation-delay:-.16s}@-webkit-keyframes sk-bouncedelay{0%,100%,80%{-webkit-transform:scale(0)}40%{-webkit-transform:scale(1)}}@keyframes sk-bouncedelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}"]
                    },] },
        ];
        /** @nocollapse */
        HorizontalBounceComponent.ctorParameters = function () { return []; };
        return HorizontalBounceComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var Spin1Component = (function () {
        function Spin1Component() {
        }
        /**
         * @return {?}
         */
        Spin1Component.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.styles.nativeElement.innerHTML = "<style>.sk-circle .sk-child:before { background-color: " + this.config.color + " }</style>";
            };
        Spin1Component.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-spin1',
                        template: "<div class=\"sk-circle\" [style.height]=\"config.size\" [style.width]=\"config.size\">\n  <div class=\"sk-circle1 sk-child\" ></div>\n  <div class=\"sk-circle2 sk-child\" ></div>\n  <div class=\"sk-circle3 sk-child\" ></div>\n  <div class=\"sk-circle4 sk-child\" ></div>\n  <div class=\"sk-circle5 sk-child\" ></div>\n  <div class=\"sk-circle6 sk-child\" ></div>\n  <div class=\"sk-circle7 sk-child\" ></div>\n  <div class=\"sk-circle8 sk-child\" ></div>\n  <div class=\"sk-circle9 sk-child\" ></div>\n  <div class=\"sk-circle10 sk-child\"></div>\n  <div class=\"sk-circle11 sk-child\"></div>\n  <div class=\"sk-circle12 sk-child\"></div>\n</div>\n<div #styles></div>",
                        styles: [".sk-circle{position:relative}.sk-circle .sk-child{width:100%;height:100%;position:absolute;left:0;top:0}.sk-circle .sk-child:before{content:'';display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleBounceDelay;animation:1.2s ease-in-out infinite both sk-circleBounceDelay}.sk-circle .sk-circle2{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-circle .sk-circle3{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-circle .sk-circle4{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-circle .sk-circle5{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-circle .sk-circle6{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-circle .sk-circle7{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-circle .sk-circle8{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-circle .sk-circle9{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-circle .sk-circle10{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-circle .sk-circle11{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-circle .sk-circle12{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-circle .sk-circle2:before{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-circle .sk-circle3:before{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-circle .sk-circle4:before{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-circle .sk-circle5:before{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-circle .sk-circle6:before{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-circle .sk-circle7:before{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-circle .sk-circle8:before{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-circle .sk-circle9:before{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-circle .sk-circle10:before{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-circle .sk-circle11:before{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-circle .sk-circle12:before{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}"]
                    },] },
        ];
        /** @nocollapse */
        Spin1Component.ctorParameters = function () { return []; };
        Spin1Component.propDecorators = {
            styles: [{ type: core.ViewChild, args: ["styles",] }]
        };
        return Spin1Component;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var Spin2Component = (function () {
        function Spin2Component() {
        }
        /**
         * @return {?}
         */
        Spin2Component.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.styles.nativeElement.innerHTML = "<style>.sk-fading-circle .sk-circle:before { background-color: " + this.config.color + " }</style>";
            };
        Spin2Component.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-spin2',
                        template: "<div class=\"sk-fading-circle\" [style.height]=\"config.size\" [style.width]=\"config.size\" >\n\t\t<div class=\"sk-circle1 sk-circle\"></div>\n\t\t<div class=\"sk-circle2 sk-circle\"></div>\n\t\t<div class=\"sk-circle3 sk-circle\"></div>\n\t\t<div class=\"sk-circle4 sk-circle\"></div>\n\t\t<div class=\"sk-circle5 sk-circle\"></div>\n\t\t<div class=\"sk-circle6 sk-circle\"></div>\n\t\t<div class=\"sk-circle7 sk-circle\"></div>\n\t\t<div class=\"sk-circle8 sk-circle\"></div>\n\t\t<div class=\"sk-circle9 sk-circle\"></div>\n\t\t<div class=\"sk-circle10 sk-circle\"></div>\n\t\t<div class=\"sk-circle11 sk-circle\"></div>\n\t\t<div class=\"sk-circle12 sk-circle\"></div>\n\t</div>\n\t<div #styles></div>",
                        styles: [".sk-fading-circle{position:relative}.sk-fading-circle .sk-circle{width:100%;height:100%;position:absolute;left:0;top:0}.sk-fading-circle .sk-circle:before{content:'';display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleFadeDelay;animation:1.2s ease-in-out infinite both sk-circleFadeDelay}.sk-fading-circle .sk-circle2{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-fading-circle .sk-circle3{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-fading-circle .sk-circle4{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-fading-circle .sk-circle5{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-fading-circle .sk-circle6{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-fading-circle .sk-circle7{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-fading-circle .sk-circle8{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-fading-circle .sk-circle9{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-fading-circle .sk-circle10{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-fading-circle .sk-circle11{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-fading-circle .sk-circle12{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-fading-circle .sk-circle2:before{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-fading-circle .sk-circle3:before{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-fading-circle .sk-circle4:before{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-fading-circle .sk-circle5:before{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-fading-circle .sk-circle6:before{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-fading-circle .sk-circle7:before{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-fading-circle .sk-circle8:before{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-fading-circle .sk-circle9:before{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-fading-circle .sk-circle10:before{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-fading-circle .sk-circle11:before{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-fading-circle .sk-circle12:before{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}@keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}"]
                    },] },
        ];
        /** @nocollapse */
        Spin2Component.ctorParameters = function () { return []; };
        Spin2Component.propDecorators = {
            styles: [{ type: core.ViewChild, args: ["styles",] }]
        };
        return Spin2Component;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var Spin3Component = (function () {
        function Spin3Component() {
        }
        /**
         * @return {?}
         */
        Spin3Component.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        Spin3Component.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-spin3',
                        template: "<div [style.height]=\"config.size\" [style.width]=\"config.size\" class=\"lds-ring\">\n\t<div [style.height]=\"config.size\" [style.width]=\"config.size\" [style.borderTopColor]=\"config.color\"></div>\n\t<div [style.height]=\"config.size\" [style.width]=\"config.size\" [style.borderTopColor]=\"config.color\"></div>\n\t<div [style.height]=\"config.size\" [style.width]=\"config.size\" [style.borderTopColor]=\"config.color\"></div>\n\t<div [style.height]=\"config.size\" [style.width]=\"config.size\" [style.borderTopColor]=\"config.color\"></div>\n</div>",
                        styles: [".lds-ring div{box-sizing:border-box;display:block;position:absolute;border:.3rem solid transparent;border-radius:50%;-webkit-animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-ring;animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-ring}.lds-ring div:nth-child(1){-webkit-animation-delay:-.45s;animation-delay:-.45s}.lds-ring div:nth-child(2){-webkit-animation-delay:-.3s;animation-delay:-.3s}.lds-ring div:nth-child(3){-webkit-animation-delay:-.15s;animation-delay:-.15s}@-webkit-keyframes lds-ring{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes lds-ring{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                    },] },
        ];
        /** @nocollapse */
        Spin3Component.ctorParameters = function () { return []; };
        return Spin3Component;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var SpinnerAnimationDirective = (function () {
        function SpinnerAnimationDirective(viewContainerRef, componentFactoryResolver) {
            this.viewContainerRef = viewContainerRef;
            this.componentFactoryResolver = componentFactoryResolver;
        }
        Object.defineProperty(SpinnerAnimationDirective.prototype, "config", {
            set: /**
             * @param {?} conf
             * @return {?}
             */ function (conf) {
                this.init(conf);
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @param {?} conf
         * @return {?}
         */
        SpinnerAnimationDirective.prototype.init = /**
         * @param {?} conf
         * @return {?}
         */
            function (conf) {
                var /** @type {?} */ componenet = BlinkComponent;
                if (conf.animation == SPINNER_ANIMATIONS.blink) {
                    // blink
                    componenet = BlinkComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.fb_card) {
                    // fb_card
                    componenet = FbCardComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.plate_rotate) {
                    componenet = PlateRotateComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.double_bounce) {
                    componenet = DoubleBounceComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.folding_cube) {
                    componenet = FoldingCubeComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.bars) {
                    componenet = BarsComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.rotating_dots) {
                    componenet = RotatingDotsComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.horizontal_bounce) {
                    componenet = HorizontalBounceComponent;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.spin_1) {
                    componenet = Spin1Component;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.spin_2) {
                    componenet = Spin2Component;
                }
                else if (conf.animation == SPINNER_ANIMATIONS.spin_3) {
                    componenet = Spin3Component;
                } /* else if(conf.animation==SPINNER_ANIMATIONS.spin_4) {
                            componenet = Spin4Component;
                        } */
                var /** @type {?} */ componentFactory = this.componentFactoryResolver.resolveComponentFactory(componenet);
                this.viewContainerRef.clear();
                var /** @type {?} */ componentRef = this.viewContainerRef.createComponent(componentFactory);
                (componentRef.instance).config = conf;
            };
        SpinnerAnimationDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[spinner-animation]'
                    },] },
        ];
        /** @nocollapse */
        SpinnerAnimationDirective.ctorParameters = function () {
            return [
                { type: core.ViewContainerRef },
                { type: core.ComponentFactoryResolver }
            ];
        };
        SpinnerAnimationDirective.propDecorators = {
            config: [{ type: core.Input, args: ['spinner-animation',] }]
        };
        return SpinnerAnimationDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var Spin4Component = (function () {
        function Spin4Component() {
        }
        /**
         * @return {?}
         */
        Spin4Component.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        Spin4Component.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-spin4',
                        template: "",
                        styles: [""]
                    },] },
        ];
        /** @nocollapse */
        Spin4Component.ctorParameters = function () { return []; };
        return Spin4Component;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var NgxSpinnerModule = (function () {
        function NgxSpinnerModule() {
        }
        /**
         * @param {?=} conf
         * @return {?}
         */
        NgxSpinnerModule.config = /**
         * @param {?=} conf
         * @return {?}
         */
            function (conf) {
                if (conf === void 0) {
                    conf = {};
                }
                NgxSpinnerService.globalSpinnerConf = Object.assign(conf);
                return this;
            };
        NgxSpinnerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        entryComponents: [
                            Spin4Component, Spin3Component, Spin2Component, Spin1Component, HorizontalBounceComponent, BlinkComponent, FbCardComponent, PlateRotateComponent, DoubleBounceComponent, FoldingCubeComponent, BarsComponent, RotatingDotsComponent
                        ],
                        declarations: [
                            SpinnerAnimationDirective,
                            NgxSpinner,
                            BlinkComponent, FbCardComponent, PlateRotateComponent, DoubleBounceComponent, FoldingCubeComponent, BarsComponent, RotatingDotsComponent, HorizontalBounceComponent, Spin1Component, Spin2Component, Spin3Component, Spin4Component
                        ],
                        exports: [NgxSpinner],
                        providers: [NgxSpinnerService]
                    },] },
        ];
        return NgxSpinnerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.NgxSpinner = NgxSpinner;
    exports.SPINNER_ANIMATIONS = SPINNER_ANIMATIONS;
    exports.SPINNER_PLACEMENT = SPINNER_PLACEMENT;
    exports.NgxSpinnerModule = NgxSpinnerModule;
    exports.ɵk = BarsComponent;
    exports.ɵf = BlinkComponent;
    exports.ɵi = DoubleBounceComponent;
    exports.ɵg = FbCardComponent;
    exports.ɵj = FoldingCubeComponent;
    exports.ɵe = HorizontalBounceComponent;
    exports.ɵn = NgxSpinnerService;
    exports.ɵh = PlateRotateComponent;
    exports.ɵl = RotatingDotsComponent;
    exports.ɵd = Spin1Component;
    exports.ɵc = Spin2Component;
    exports.ɵb = Spin3Component;
    exports.ɵa = Spin4Component;
    exports.ɵm = SpinnerAnimationDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
