// @hardpool/ngx-spinner/esm5/src/app/ngx-spinner/folding-cube/folding-cube.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, ViewChild, ElementRef } from '@angular/core';
var FoldingCubeComponent = /** @class */ (function () {
    function FoldingCubeComponent() {
    }
    /**
     * @return {?}
     */
    FoldingCubeComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.styles.nativeElement.innerHTML = "<style>.sk-folding-cube .sk-cube:before { background-color: " + this.config.color + " }</style>";
    };
    FoldingCubeComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-folding-cube',
                    template: "<div class=\"sk-folding-cube\" [style.height]=\"config.size\" [style.width]=\"config.size\">\n  <div class=\"sk-cube1 sk-cube\"></div>\n  <div class=\"sk-cube2 sk-cube\"></div>\n  <div class=\"sk-cube4 sk-cube\"></div>\n  <div class=\"sk-cube3 sk-cube\"></div>\n</div>\n<div #styles></div>",
                    styles: [".sk-folding-cube{position:relative;-webkit-transform:rotateZ(45deg);transform:rotateZ(45deg)}.sk-folding-cube .sk-cube{float:left;width:50%;height:50%;position:relative;-webkit-transform:scale(1.1);transform:scale(1.1)}.sk-folding-cube .sk-cube:before{content:'';position:absolute;top:0;left:0;width:100%;height:100%;-webkit-animation:2.4s linear infinite both sk-foldCubeAngle;animation:2.4s linear infinite both sk-foldCubeAngle;-webkit-transform-origin:100% 100%;transform-origin:100% 100%}.sk-folding-cube .sk-cube2{-webkit-transform:scale(1.1) rotateZ(90deg);transform:scale(1.1) rotateZ(90deg)}.sk-folding-cube .sk-cube3{-webkit-transform:scale(1.1) rotateZ(180deg);transform:scale(1.1) rotateZ(180deg)}.sk-folding-cube .sk-cube4{-webkit-transform:scale(1.1) rotateZ(270deg);transform:scale(1.1) rotateZ(270deg)}.sk-folding-cube .sk-cube2:before{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-folding-cube .sk-cube3:before{-webkit-animation-delay:.6s;animation-delay:.6s}.sk-folding-cube .sk-cube4:before{-webkit-animation-delay:.9s;animation-delay:.9s}@-webkit-keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}@keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}"]
                },] },
    ];
    /** @nocollapse */
    FoldingCubeComponent.ctorParameters = function () { return []; };
    FoldingCubeComponent.propDecorators = {
        styles: [{ type: ViewChild, args: ["styles",] }]
    };
    return FoldingCubeComponent;
}());
export { FoldingCubeComponent };
function FoldingCubeComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    FoldingCubeComponent.prototype.styles;
    /** @type {?} */
    FoldingCubeComponent.prototype.config;
}
