// @hardpool/ngx-spinner/esm2015/src/app/ngx-spinner/spin3/spin3.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
export class Spin3Component {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
Spin3Component.decorators = [
    { type: Component, args: [{
                selector: 'app-spin3',
                template: `<div [style.height]="config.size" [style.width]="config.size" class="lds-ring">
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
</div>`,
                styles: [`.lds-ring div{box-sizing:border-box;display:block;position:absolute;border:.3rem solid transparent;border-radius:50%;-webkit-animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-ring;animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-ring}.lds-ring div:nth-child(1){-webkit-animation-delay:-.45s;animation-delay:-.45s}.lds-ring div:nth-child(2){-webkit-animation-delay:-.3s;animation-delay:-.3s}.lds-ring div:nth-child(3){-webkit-animation-delay:-.15s;animation-delay:-.15s}@-webkit-keyframes lds-ring{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes lds-ring{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}`]
            },] },
];
/** @nocollapse */
Spin3Component.ctorParameters = () => [];
function Spin3Component_tsickle_Closure_declarations() {
    /** @type {?} */
    Spin3Component.prototype.config;
}
