// @hardpool/ngx-spinner/esm5/src/app/ngx-spinner/ngx-spinner.types.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function ISpinnerConfig() { }
function ISpinnerConfig_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ISpinnerConfig.prototype.placement;
    /** @type {?|undefined} */
    ISpinnerConfig.prototype.animation;
    /** @type {?|undefined} */
    ISpinnerConfig.prototype.size;
    /** @type {?|undefined} */
    ISpinnerConfig.prototype.bgColor;
    /** @type {?|undefined} */
    ISpinnerConfig.prototype.color;
    /** @type {?|undefined} */
    ISpinnerConfig.prototype.secondaryColor;
    /** @type {?|undefined} */
    ISpinnerConfig.prototype.tertiaryColor;
}
/**
 * @record
 */
export function ISpinnerAnimation() { }
function ISpinnerAnimation_tsickle_Closure_declarations() {
    /** @type {?} */
    ISpinnerAnimation.prototype.config;
}
/** @enum {string} */
var SPINNER_PLACEMENT = {
    block_window: "block_window",
    block_ui: "block_ui",
    inplace: "inplace",
};
export { SPINNER_PLACEMENT };
/** @enum {number} */
var SPINNER_ANIMATIONS = {
    blink: 0,
    fb_card: 1,
    plate_rotate: 2,
    double_bounce: 3,
    folding_cube: 4,
    bars: 5,
    rotating_dots: 6,
    horizontal_bounce: 7,
    spin_1: 8,
    spin_2: 9,
    spin_3: 10,
};
export { SPINNER_ANIMATIONS };
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.blink] = "blink";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.fb_card] = "fb_card";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.plate_rotate] = "plate_rotate";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.double_bounce] = "double_bounce";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.folding_cube] = "folding_cube";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.bars] = "bars";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.rotating_dots] = "rotating_dots";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.horizontal_bounce] = "horizontal_bounce";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_1] = "spin_1";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_2] = "spin_2";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_3] = "spin_3";
export var /** @type {?} */ DEFAULT_CONFIG = {
    placement: SPINNER_PLACEMENT.inplace,
    animation: SPINNER_ANIMATIONS.blink,
    size: "2rem",
    bgColor: "rgba(255,255,255,0.8)",
    color: "#e46521",
    secondaryColor: "#1574b3",
    tertiaryColor: "#c31a4f"
};
