// @hardpool/ngx-spinner/esm5/src/app/ngx-spinner/spinner-animation.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, Input, ViewContainerRef, ComponentFactoryResolver } from '@angular/core';
import { SPINNER_ANIMATIONS } from './ngx-spinner.types';
import { BlinkComponent } from './blink/blink.component';
import { FbCardComponent } from './fb-card/fb-card.component';
import { PlateRotateComponent } from './plate-rotate/plate-rotate.component';
import { DoubleBounceComponent } from './double-bounce/double-bounce.component';
import { FoldingCubeComponent } from './folding-cube/folding-cube.component';
import { BarsComponent } from './bars/bars.component';
import { RotatingDotsComponent } from './rotating-dots/rotating-dots.component';
import { HorizontalBounceComponent } from './horizontal-bounce/horizontal-bounce.component';
import { Spin1Component } from './spin1/spin1.component';
import { Spin2Component } from './spin2/spin2.component';
import { Spin3Component } from './spin3/spin3.component';
var SpinnerAnimationDirective = /** @class */ (function () {
    function SpinnerAnimationDirective(viewContainerRef, componentFactoryResolver) {
        this.viewContainerRef = viewContainerRef;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    Object.defineProperty(SpinnerAnimationDirective.prototype, "config", {
        set: /**
         * @param {?} conf
         * @return {?}
         */
        function (conf) {
            this.init(conf);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} conf
     * @return {?}
     */
    SpinnerAnimationDirective.prototype.init = /**
     * @param {?} conf
     * @return {?}
     */
    function (conf) {
        var /** @type {?} */ componenet = BlinkComponent;
        if (conf.animation == SPINNER_ANIMATIONS.blink) {
            // blink
            componenet = BlinkComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.fb_card) {
            // fb_card
            componenet = FbCardComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.plate_rotate) {
            componenet = PlateRotateComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.double_bounce) {
            componenet = DoubleBounceComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.folding_cube) {
            componenet = FoldingCubeComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.bars) {
            componenet = BarsComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.rotating_dots) {
            componenet = RotatingDotsComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.horizontal_bounce) {
            componenet = HorizontalBounceComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.spin_1) {
            componenet = Spin1Component;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.spin_2) {
            componenet = Spin2Component;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.spin_3) {
            componenet = Spin3Component;
        } /* else if(conf.animation==SPINNER_ANIMATIONS.spin_4) {
                    componenet = Spin4Component;
                } */
        var /** @type {?} */ componentFactory = this.componentFactoryResolver.resolveComponentFactory(componenet);
        this.viewContainerRef.clear();
        var /** @type {?} */ componentRef = this.viewContainerRef.createComponent(componentFactory);
        (componentRef.instance).config = conf;
    };
    SpinnerAnimationDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[spinner-animation]'
                },] },
    ];
    /** @nocollapse */
    SpinnerAnimationDirective.ctorParameters = function () { return [
        { type: ViewContainerRef },
        { type: ComponentFactoryResolver }
    ]; };
    SpinnerAnimationDirective.propDecorators = {
        config: [{ type: Input, args: ['spinner-animation',] }]
    };
    return SpinnerAnimationDirective;
}());
export { SpinnerAnimationDirective };
function SpinnerAnimationDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    SpinnerAnimationDirective.prototype.viewContainerRef;
    /** @type {?} */
    SpinnerAnimationDirective.prototype.componentFactoryResolver;
}
