// @hardpool/ngx-spinner/esm5/src/app/ngx-spinner/double-bounce/double-bounce.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
var DoubleBounceComponent = /** @class */ (function () {
    function DoubleBounceComponent() {
    }
    /**
     * @return {?}
     */
    DoubleBounceComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    DoubleBounceComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-double-bounce',
                    template: "<div class=\"spinner\" [style.height]=\"config.size\" [style.width]=\"config.size\">\n  <div class=\"double-bounce1\" [style.background]=\"config.color\"></div>\n  <div class=\"double-bounce2\" [style.background]=\"config.secondaryColor\"></div>\n</div>",
                    styles: [".spinner{position:relative}.double-bounce1,.double-bounce2{width:100%;height:100%;border-radius:50%;opacity:.6;position:absolute;top:0;left:0;-webkit-animation:2s ease-in-out infinite sk-bounce;animation:2s ease-in-out infinite sk-bounce}.double-bounce2{-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-bounce{0%,100%{-webkit-transform:scale(0)}50%{-webkit-transform:scale(1)}}@keyframes sk-bounce{0%,100%{transform:scale(0);-webkit-transform:scale(0)}50%{transform:scale(1);-webkit-transform:scale(1)}}"]
                },] },
    ];
    /** @nocollapse */
    DoubleBounceComponent.ctorParameters = function () { return []; };
    return DoubleBounceComponent;
}());
export { DoubleBounceComponent };
function DoubleBounceComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    DoubleBounceComponent.prototype.config;
}
