// @hardpool/ngx-spinner/fesm2015/hardpool-ngx-spinner.js
import { Injectable, Component, Input, ViewChild, Directive, ViewContainerRef, ComponentFactoryResolver, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxSpinnerService {
    constructor() { }
}
NgxSpinnerService.globalSpinnerConf = {};
NgxSpinnerService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
NgxSpinnerService.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/** @enum {string} */
const SPINNER_PLACEMENT = {
    block_window: "block_window",
    block_ui: "block_ui",
    inplace: "inplace",
};
/** @enum {number} */
const SPINNER_ANIMATIONS = {
    blink: 0,
    fb_card: 1,
    plate_rotate: 2,
    double_bounce: 3,
    folding_cube: 4,
    bars: 5,
    rotating_dots: 6,
    horizontal_bounce: 7,
    spin_1: 8,
    spin_2: 9,
    spin_3: 10,
};
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.blink] = "blink";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.fb_card] = "fb_card";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.plate_rotate] = "plate_rotate";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.double_bounce] = "double_bounce";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.folding_cube] = "folding_cube";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.bars] = "bars";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.rotating_dots] = "rotating_dots";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.horizontal_bounce] = "horizontal_bounce";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_1] = "spin_1";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_2] = "spin_2";
SPINNER_ANIMATIONS[SPINNER_ANIMATIONS.spin_3] = "spin_3";
const /** @type {?} */ DEFAULT_CONFIG = {
    placement: SPINNER_PLACEMENT.inplace,
    animation: SPINNER_ANIMATIONS.blink,
    size: "2rem",
    bgColor: "rgba(255,255,255,0.8)",
    color: "#e46521",
    secondaryColor: "#1574b3",
    tertiaryColor: "#c31a4f"
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxSpinner {
    /**
     * @param {?} ngxSpinnerService
     */
    constructor(ngxSpinnerService) {
        this.ngxSpinnerService = ngxSpinnerService;
        this._config = {};
        // assign default config
        this.assignConfig(DEFAULT_CONFIG);
        // override wtih globle config
        this.assignConfig(NgxSpinnerService.globalSpinnerConf);
        // hide spinner as default
        this._isVisible = false;
    }
    /**
     * @param {?} condition
     * @return {?}
     */
    set visible(condition) {
        if (condition)
            this.show();
        else
            this.hide();
    }
    /**
     * @return {?}
     */
    get visible() {
        return this._isVisible;
    }
    /**
     * @param {?} conf
     * @return {?}
     */
    set config(conf) {
        // override with spinner config
        this.assignConfig(conf);
    }
    /**
     * @return {?}
     */
    get config() {
        return this._config;
    }
    /**
     * @return {?}
     */
    show() {
        this._isVisible = true;
    }
    /**
     * @return {?}
     */
    hide() {
        this._isVisible = false;
    }
    /**
     * @return {?}
     */
    toggle() {
        this._isVisible = !this._isVisible;
    }
    /**
     * @param {?} conf
     * @return {?}
     */
    assignConfig(conf) {
        let /** @type {?} */ cnf = Object.assign(conf);
        for (let /** @type {?} */ prop in cnf) {
            if (cnf.hasOwnProperty(prop))
                this._config[prop] = cnf[prop];
        }
    }
}
NgxSpinner.decorators = [
    { type: Component, args: [{
                selector: 'ngx-spinner',
                template: `<!-- spinner -->
<ng-template *ngIf="visible" [spinner-animation]="config"></ng-template>`,
                styles: [`:host.block_window{align-items:center;justify-content:center;position:fixed;display:flex;width:100%;height:100%;top:0;left:0;z-index:10002}:host.block_ui{align-items:center;justify-content:center;position:absolute;display:flex;width:100%;height:100%;top:0;left:0;z-index:10001}:host.inplace{position:absolute;background:0 0!important;display:flex;width:100%;height:100%;top:0;left:0;align-items:center;justify-content:start}`],
                host: { '[class]': 'visible?config.placement:""', '[style.background]': 'visible?config.bgColor:""' }
            },] },
];
/** @nocollapse */
NgxSpinner.ctorParameters = () => [
    { type: NgxSpinnerService }
];
NgxSpinner.propDecorators = {
    visible: [{ type: Input }],
    config: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class BlinkComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
BlinkComponent.decorators = [
    { type: Component, args: [{
                selector: 'spinner-blink',
                template: `<div class="sk-scaleout" [style.height]="config.size" [style.width]="config.size" [style.background]="config.color">
</div>`,
                styles: [`.sk-scaleout{margin:0;border-radius:100%;-webkit-animation:1s ease-in-out infinite sk-scaleout;animation:1s ease-in-out infinite sk-scaleout}@-webkit-keyframes sk-scaleout{0%{-webkit-transform:scale(0)}100%{-webkit-transform:scale(1);opacity:0}}@keyframes sk-scaleout{0%{-webkit-transform:scale(0);transform:scale(0)}100%{-webkit-transform:scale(1);transform:scale(1);opacity:0}}`]
            },] },
];
/** @nocollapse */
BlinkComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class FbCardComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
FbCardComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fb-card',
                template: `<div class="FB-Loading-Card" [style.height]="config.size" [style.width]="config.size">
	<div>
	 <div [style.background]="config.color"></div>
	 <div [style.background]="config.color"></div>
	 <div [style.background]="config.color"></div>
	</div>
	<div [style.background]="config.color"></div>
	<div [style.background]="config.color"></div>
	<div [style.background]="config.color"></div>
 </div>`,
                styles: [`@charset "UTF-8";.FB-Loading-Card{display:block;padding:12px}.FB-Loading-Card>div:nth-child(1){width:100%;height:30px}.FB-Loading-Card>div>div:nth-child(1){position:relative;left:0;height:40px;width:40px;-webkit-animation-name:shine;animation-name:shine;-webkit-animation-duration:2s;animation-duration:2s;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite;-webkit-animation-timing-function:linear;animation-timing-function:linear}.FB-Loading-Card>div:nth-child(2),.FB-Loading-Card>div:nth-child(3),.FB-Loading-Card>div:nth-child(4),.FB-Loading-Card>div>div:nth-child(2),.FB-Loading-Card>div>div:nth-child(3){position:relative;height:6px;-webkit-animation-name:shine;animation-name:shine;-webkit-animation-duration:2s;animation-duration:2s;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite;-webkit-animation-timing-function:linear;animation-timing-function:linear}@-webkit-keyframes shine{from{transition:none}to{background-color:#f6f7f8;transition:all .3s ease-out}}@keyframes shine{from{transition:none}to{background-color:#f6f7f8;transition:all .3s ease-out}}.FB-Loading-Card>div>div:nth-child(2){left:50px;top:-33px;width:25%}.FB-Loading-Card>div>div:nth-child(3){left:50px;top:-20px;width:15%}.FB-Loading-Card>div:nth-child(2){left:0;top:30px;width:80%}.FB-Loading-Card>div:nth-child(3){left:0;top:45px;width:90%}.FB-Loading-Card>div:nth-child(4){left:0;top:60px;width:50%}`]
            },] },
];
/** @nocollapse */
FbCardComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class PlateRotateComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
PlateRotateComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-plate-rotate',
                template: `<div class="spinner" [style.height]="config.size" [style.width]="config.size" [style.background]="config.color"	></div>`,
                styles: [`.spinner{width:40px;height:40px;-webkit-animation:1.2s ease-in-out infinite sk-rotateplane;animation:1.2s ease-in-out infinite sk-rotateplane}@-webkit-keyframes sk-rotateplane{0%{-webkit-transform:perspective(120px)}50%{-webkit-transform:perspective(120px) rotateY(180deg)}100%{-webkit-transform:perspective(120px) rotateY(180deg) rotateX(180deg)}}@keyframes sk-rotateplane{0%{transform:perspective(120px) rotateX(0) rotateY(0);-webkit-transform:perspective(120px) rotateX(0) rotateY(0)}50%{transform:perspective(120px) rotateX(-180.1deg) rotateY(0);-webkit-transform:perspective(120px) rotateX(-180.1deg) rotateY(0)}100%{transform:perspective(120px) rotateX(-180deg) rotateY(-179.9deg);-webkit-transform:perspective(120px) rotateX(-180deg) rotateY(-179.9deg)}}`]
            },] },
];
/** @nocollapse */
PlateRotateComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class DoubleBounceComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
DoubleBounceComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-double-bounce',
                template: `<div class="spinner" [style.height]="config.size" [style.width]="config.size">
  <div class="double-bounce1" [style.background]="config.color"></div>
  <div class="double-bounce2" [style.background]="config.secondaryColor"></div>
</div>`,
                styles: [`.spinner{position:relative}.double-bounce1,.double-bounce2{width:100%;height:100%;border-radius:50%;opacity:.6;position:absolute;top:0;left:0;-webkit-animation:2s ease-in-out infinite sk-bounce;animation:2s ease-in-out infinite sk-bounce}.double-bounce2{-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-bounce{0%,100%{-webkit-transform:scale(0)}50%{-webkit-transform:scale(1)}}@keyframes sk-bounce{0%,100%{transform:scale(0);-webkit-transform:scale(0)}50%{transform:scale(1);-webkit-transform:scale(1)}}`]
            },] },
];
/** @nocollapse */
DoubleBounceComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class FoldingCubeComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.styles.nativeElement.innerHTML = `<style>.sk-folding-cube .sk-cube:before { background-color: ${this.config.color} }</style>`;
    }
}
FoldingCubeComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-folding-cube',
                template: `<div class="sk-folding-cube" [style.height]="config.size" [style.width]="config.size">
  <div class="sk-cube1 sk-cube"></div>
  <div class="sk-cube2 sk-cube"></div>
  <div class="sk-cube4 sk-cube"></div>
  <div class="sk-cube3 sk-cube"></div>
</div>
<div #styles></div>`,
                styles: [`.sk-folding-cube{position:relative;-webkit-transform:rotateZ(45deg);transform:rotateZ(45deg)}.sk-folding-cube .sk-cube{float:left;width:50%;height:50%;position:relative;-webkit-transform:scale(1.1);transform:scale(1.1)}.sk-folding-cube .sk-cube:before{content:'';position:absolute;top:0;left:0;width:100%;height:100%;-webkit-animation:2.4s linear infinite both sk-foldCubeAngle;animation:2.4s linear infinite both sk-foldCubeAngle;-webkit-transform-origin:100% 100%;transform-origin:100% 100%}.sk-folding-cube .sk-cube2{-webkit-transform:scale(1.1) rotateZ(90deg);transform:scale(1.1) rotateZ(90deg)}.sk-folding-cube .sk-cube3{-webkit-transform:scale(1.1) rotateZ(180deg);transform:scale(1.1) rotateZ(180deg)}.sk-folding-cube .sk-cube4{-webkit-transform:scale(1.1) rotateZ(270deg);transform:scale(1.1) rotateZ(270deg)}.sk-folding-cube .sk-cube2:before{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-folding-cube .sk-cube3:before{-webkit-animation-delay:.6s;animation-delay:.6s}.sk-folding-cube .sk-cube4:before{-webkit-animation-delay:.9s;animation-delay:.9s}@-webkit-keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}@keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}`]
            },] },
];
/** @nocollapse */
FoldingCubeComponent.ctorParameters = () => [];
FoldingCubeComponent.propDecorators = {
    styles: [{ type: ViewChild, args: ["styles",] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class BarsComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
BarsComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-bars',
                template: `<div class="spinner" [style.height]="config.size" >
  <div class="rect1" [style.background]="config.color"></div>
  <div class="rect2" [style.background]="config.color"></div>
  <div class="rect3" [style.background]="config.color"></div>
  <div class="rect4" [style.background]="config.color"></div>
  <div class="rect5" [style.background]="config.color"></div>
</div>`,
                styles: [`.spinner{margin:0}.spinner>div{height:100%;width:.4rem;margin:0 .15rem;display:inline-block;-webkit-animation:1.2s ease-in-out infinite sk-stretchdelay;animation:1.2s ease-in-out infinite sk-stretchdelay}.spinner .rect2{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.spinner .rect3{-webkit-animation-delay:-1s;animation-delay:-1s}.spinner .rect4{-webkit-animation-delay:-.9s;animation-delay:-.9s}.spinner .rect5{-webkit-animation-delay:-.8s;animation-delay:-.8s}@-webkit-keyframes sk-stretchdelay{0%,100%,40%{-webkit-transform:scaleY(.4)}20%{-webkit-transform:scaleY(1)}}@keyframes sk-stretchdelay{0%,100%,40%{transform:scaleY(.4);-webkit-transform:scaleY(.4)}20%{transform:scaleY(1);-webkit-transform:scaleY(1)}}`]
            },] },
];
/** @nocollapse */
BarsComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class RotatingDotsComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
RotatingDotsComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-rotating-dots',
                template: `<div class="spinner" [style.height]="config.size" [style.width]="config.size">
  <div class="dot1" [style.background]="config.color"></div>
  <div class="dot2" [style.background]="config.color"></div>
</div>`,
                styles: [`.spinner{position:relative;text-align:center;-webkit-animation:2s linear infinite sk-rotate;animation:2s linear infinite sk-rotate}.dot1,.dot2{width:60%;height:60%;display:inline-block;position:absolute;top:0;border-radius:100%;-webkit-animation:2s ease-in-out infinite sk-bounce;animation:2s ease-in-out infinite sk-bounce}.dot2{top:auto;bottom:0;-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-rotate{100%{-webkit-transform:rotate(360deg)}}@keyframes sk-rotate{100%{transform:rotate(360deg);-webkit-transform:rotate(360deg)}}@-webkit-keyframes sk-bounce{0%,100%{-webkit-transform:scale(0)}50%{-webkit-transform:scale(1)}}@keyframes sk-bounce{0%,100%{transform:scale(0);-webkit-transform:scale(0)}50%{transform:scale(1);-webkit-transform:scale(1)}}`]
            },] },
];
/** @nocollapse */
RotatingDotsComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class HorizontalBounceComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
HorizontalBounceComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-horizontal-bounce',
                template: `<div class="spinner">
  <div class="bounce1" [style.height]="config.size" [style.width]="config.size" [style.background]="config.color"></div>
  <div class="bounce2" [style.height]="config.size" [style.width]="config.size" [style.background]="config.color"></div>
  <div class="bounce3" [style.height]="config.size" [style.width]="config.size" [style.background]="config.color"></div>
</div>`,
                styles: [`.spinner{text-align:center}.spinner>div{margin:0 .15rem;border-radius:100%;display:inline-block;-webkit-animation:1.4s ease-in-out infinite both sk-bouncedelay;animation:1.4s ease-in-out infinite both sk-bouncedelay}.spinner .bounce1{-webkit-animation-delay:-.32s;animation-delay:-.32s}.spinner .bounce2{-webkit-animation-delay:-.16s;animation-delay:-.16s}@-webkit-keyframes sk-bouncedelay{0%,100%,80%{-webkit-transform:scale(0)}40%{-webkit-transform:scale(1)}}@keyframes sk-bouncedelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}`]
            },] },
];
/** @nocollapse */
HorizontalBounceComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class Spin1Component {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.styles.nativeElement.innerHTML = `<style>.sk-circle .sk-child:before { background-color: ${this.config.color} }</style>`;
    }
}
Spin1Component.decorators = [
    { type: Component, args: [{
                selector: 'app-spin1',
                template: `<div class="sk-circle" [style.height]="config.size" [style.width]="config.size">
  <div class="sk-circle1 sk-child" ></div>
  <div class="sk-circle2 sk-child" ></div>
  <div class="sk-circle3 sk-child" ></div>
  <div class="sk-circle4 sk-child" ></div>
  <div class="sk-circle5 sk-child" ></div>
  <div class="sk-circle6 sk-child" ></div>
  <div class="sk-circle7 sk-child" ></div>
  <div class="sk-circle8 sk-child" ></div>
  <div class="sk-circle9 sk-child" ></div>
  <div class="sk-circle10 sk-child"></div>
  <div class="sk-circle11 sk-child"></div>
  <div class="sk-circle12 sk-child"></div>
</div>
<div #styles></div>`,
                styles: [`.sk-circle{position:relative}.sk-circle .sk-child{width:100%;height:100%;position:absolute;left:0;top:0}.sk-circle .sk-child:before{content:'';display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleBounceDelay;animation:1.2s ease-in-out infinite both sk-circleBounceDelay}.sk-circle .sk-circle2{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-circle .sk-circle3{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-circle .sk-circle4{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-circle .sk-circle5{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-circle .sk-circle6{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-circle .sk-circle7{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-circle .sk-circle8{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-circle .sk-circle9{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-circle .sk-circle10{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-circle .sk-circle11{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-circle .sk-circle12{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-circle .sk-circle2:before{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-circle .sk-circle3:before{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-circle .sk-circle4:before{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-circle .sk-circle5:before{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-circle .sk-circle6:before{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-circle .sk-circle7:before{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-circle .sk-circle8:before{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-circle .sk-circle9:before{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-circle .sk-circle10:before{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-circle .sk-circle11:before{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-circle .sk-circle12:before{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}`]
            },] },
];
/** @nocollapse */
Spin1Component.ctorParameters = () => [];
Spin1Component.propDecorators = {
    styles: [{ type: ViewChild, args: ["styles",] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class Spin2Component {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.styles.nativeElement.innerHTML = `<style>.sk-fading-circle .sk-circle:before { background-color: ${this.config.color} }</style>`;
    }
}
Spin2Component.decorators = [
    { type: Component, args: [{
                selector: 'app-spin2',
                template: `<div class="sk-fading-circle" [style.height]="config.size" [style.width]="config.size" >
		<div class="sk-circle1 sk-circle"></div>
		<div class="sk-circle2 sk-circle"></div>
		<div class="sk-circle3 sk-circle"></div>
		<div class="sk-circle4 sk-circle"></div>
		<div class="sk-circle5 sk-circle"></div>
		<div class="sk-circle6 sk-circle"></div>
		<div class="sk-circle7 sk-circle"></div>
		<div class="sk-circle8 sk-circle"></div>
		<div class="sk-circle9 sk-circle"></div>
		<div class="sk-circle10 sk-circle"></div>
		<div class="sk-circle11 sk-circle"></div>
		<div class="sk-circle12 sk-circle"></div>
	</div>
	<div #styles></div>`,
                styles: [`.sk-fading-circle{position:relative}.sk-fading-circle .sk-circle{width:100%;height:100%;position:absolute;left:0;top:0}.sk-fading-circle .sk-circle:before{content:'';display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleFadeDelay;animation:1.2s ease-in-out infinite both sk-circleFadeDelay}.sk-fading-circle .sk-circle2{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-fading-circle .sk-circle3{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-fading-circle .sk-circle4{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-fading-circle .sk-circle5{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-fading-circle .sk-circle6{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-fading-circle .sk-circle7{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-fading-circle .sk-circle8{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-fading-circle .sk-circle9{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-fading-circle .sk-circle10{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-fading-circle .sk-circle11{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-fading-circle .sk-circle12{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-fading-circle .sk-circle2:before{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-fading-circle .sk-circle3:before{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-fading-circle .sk-circle4:before{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-fading-circle .sk-circle5:before{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-fading-circle .sk-circle6:before{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-fading-circle .sk-circle7:before{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-fading-circle .sk-circle8:before{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-fading-circle .sk-circle9:before{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-fading-circle .sk-circle10:before{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-fading-circle .sk-circle11:before{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-fading-circle .sk-circle12:before{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}@keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}`]
            },] },
];
/** @nocollapse */
Spin2Component.ctorParameters = () => [];
Spin2Component.propDecorators = {
    styles: [{ type: ViewChild, args: ["styles",] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class Spin3Component {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
Spin3Component.decorators = [
    { type: Component, args: [{
                selector: 'app-spin3',
                template: `<div [style.height]="config.size" [style.width]="config.size" class="lds-ring">
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
	<div [style.height]="config.size" [style.width]="config.size" [style.borderTopColor]="config.color"></div>
</div>`,
                styles: [`.lds-ring div{box-sizing:border-box;display:block;position:absolute;border:.3rem solid transparent;border-radius:50%;-webkit-animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-ring;animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-ring}.lds-ring div:nth-child(1){-webkit-animation-delay:-.45s;animation-delay:-.45s}.lds-ring div:nth-child(2){-webkit-animation-delay:-.3s;animation-delay:-.3s}.lds-ring div:nth-child(3){-webkit-animation-delay:-.15s;animation-delay:-.15s}@-webkit-keyframes lds-ring{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes lds-ring{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}`]
            },] },
];
/** @nocollapse */
Spin3Component.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class SpinnerAnimationDirective {
    /**
     * @param {?} viewContainerRef
     * @param {?} componentFactoryResolver
     */
    constructor(viewContainerRef, componentFactoryResolver) {
        this.viewContainerRef = viewContainerRef;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    /**
     * @param {?} conf
     * @return {?}
     */
    set config(conf) {
        this.init(conf);
    }
    /**
     * @param {?} conf
     * @return {?}
     */
    init(conf) {
        let /** @type {?} */ componenet = BlinkComponent;
        if (conf.animation == SPINNER_ANIMATIONS.blink) {
            // blink
            componenet = BlinkComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.fb_card) {
            // fb_card
            componenet = FbCardComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.plate_rotate) {
            componenet = PlateRotateComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.double_bounce) {
            componenet = DoubleBounceComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.folding_cube) {
            componenet = FoldingCubeComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.bars) {
            componenet = BarsComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.rotating_dots) {
            componenet = RotatingDotsComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.horizontal_bounce) {
            componenet = HorizontalBounceComponent;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.spin_1) {
            componenet = Spin1Component;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.spin_2) {
            componenet = Spin2Component;
        }
        else if (conf.animation == SPINNER_ANIMATIONS.spin_3) {
            componenet = Spin3Component;
        } /* else if(conf.animation==SPINNER_ANIMATIONS.spin_4) {
                    componenet = Spin4Component;
                } */
        let /** @type {?} */ componentFactory = this.componentFactoryResolver.resolveComponentFactory(componenet);
        this.viewContainerRef.clear();
        let /** @type {?} */ componentRef = this.viewContainerRef.createComponent(componentFactory);
        (componentRef.instance).config = conf;
    }
}
SpinnerAnimationDirective.decorators = [
    { type: Directive, args: [{
                selector: '[spinner-animation]'
            },] },
];
/** @nocollapse */
SpinnerAnimationDirective.ctorParameters = () => [
    { type: ViewContainerRef },
    { type: ComponentFactoryResolver }
];
SpinnerAnimationDirective.propDecorators = {
    config: [{ type: Input, args: ['spinner-animation',] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class Spin4Component {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
Spin4Component.decorators = [
    { type: Component, args: [{
                selector: 'app-spin4',
                template: ``,
                styles: [``]
            },] },
];
/** @nocollapse */
Spin4Component.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxSpinnerModule {
    /**
     * @param {?=} conf
     * @return {?}
     */
    static config(conf = {}) {
        NgxSpinnerService.globalSpinnerConf = Object.assign(conf);
        return this;
    }
}
NgxSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                entryComponents: [
                    Spin4Component, Spin3Component, Spin2Component, Spin1Component, HorizontalBounceComponent, BlinkComponent, FbCardComponent, PlateRotateComponent, DoubleBounceComponent, FoldingCubeComponent, BarsComponent, RotatingDotsComponent
                ],
                declarations: [
                    SpinnerAnimationDirective,
                    NgxSpinner,
                    BlinkComponent, FbCardComponent, PlateRotateComponent, DoubleBounceComponent, FoldingCubeComponent, BarsComponent, RotatingDotsComponent, HorizontalBounceComponent, Spin1Component, Spin2Component, Spin3Component, Spin4Component
                ],
                exports: [NgxSpinner],
                providers: [NgxSpinnerService]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxSpinner, SPINNER_ANIMATIONS, SPINNER_PLACEMENT, NgxSpinnerModule, BarsComponent as ɵk, BlinkComponent as ɵf, DoubleBounceComponent as ɵi, FbCardComponent as ɵg, FoldingCubeComponent as ɵj, HorizontalBounceComponent as ɵe, NgxSpinnerService as ɵn, PlateRotateComponent as ɵh, RotatingDotsComponent as ɵl, Spin1Component as ɵd, Spin2Component as ɵc, Spin3Component as ɵb, Spin4Component as ɵa, SpinnerAnimationDirective as ɵm };
