// @hardpool/ngx-spinner/esm2015/src/app/ngx-spinner/spin1/spin1.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, ViewChild, ElementRef } from '@angular/core';
export class Spin1Component {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.styles.nativeElement.innerHTML = `<style>.sk-circle .sk-child:before { background-color: ${this.config.color} }</style>`;
    }
}
Spin1Component.decorators = [
    { type: Component, args: [{
                selector: 'app-spin1',
                template: `<div class="sk-circle" [style.height]="config.size" [style.width]="config.size">
  <div class="sk-circle1 sk-child" ></div>
  <div class="sk-circle2 sk-child" ></div>
  <div class="sk-circle3 sk-child" ></div>
  <div class="sk-circle4 sk-child" ></div>
  <div class="sk-circle5 sk-child" ></div>
  <div class="sk-circle6 sk-child" ></div>
  <div class="sk-circle7 sk-child" ></div>
  <div class="sk-circle8 sk-child" ></div>
  <div class="sk-circle9 sk-child" ></div>
  <div class="sk-circle10 sk-child"></div>
  <div class="sk-circle11 sk-child"></div>
  <div class="sk-circle12 sk-child"></div>
</div>
<div #styles></div>`,
                styles: [`.sk-circle{position:relative}.sk-circle .sk-child{width:100%;height:100%;position:absolute;left:0;top:0}.sk-circle .sk-child:before{content:'';display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleBounceDelay;animation:1.2s ease-in-out infinite both sk-circleBounceDelay}.sk-circle .sk-circle2{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-circle .sk-circle3{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-circle .sk-circle4{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-circle .sk-circle5{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-circle .sk-circle6{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-circle .sk-circle7{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-circle .sk-circle8{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-circle .sk-circle9{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-circle .sk-circle10{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-circle .sk-circle11{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-circle .sk-circle12{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-circle .sk-circle2:before{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-circle .sk-circle3:before{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-circle .sk-circle4:before{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-circle .sk-circle5:before{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-circle .sk-circle6:before{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-circle .sk-circle7:before{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-circle .sk-circle8:before{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-circle .sk-circle9:before{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-circle .sk-circle10:before{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-circle .sk-circle11:before{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-circle .sk-circle12:before{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}`]
            },] },
];
/** @nocollapse */
Spin1Component.ctorParameters = () => [];
Spin1Component.propDecorators = {
    styles: [{ type: ViewChild, args: ["styles",] }]
};
function Spin1Component_tsickle_Closure_declarations() {
    /** @type {?} */
    Spin1Component.prototype.styles;
    /** @type {?} */
    Spin1Component.prototype.config;
}
