// @hardpool/ngx-spinner/esm2015/src/app/ngx-spinner/ngx-spinner/ngx-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { NgxSpinnerService } from './ngx-spinner.service';
import { DEFAULT_CONFIG } from '../ngx-spinner.types';
export class NgxSpinner {
    /**
     * @param {?} ngxSpinnerService
     */
    constructor(ngxSpinnerService) {
        this.ngxSpinnerService = ngxSpinnerService;
        this._config = {};
        // assign default config
        this.assignConfig(DEFAULT_CONFIG);
        // override wtih globle config
        this.assignConfig(NgxSpinnerService.globalSpinnerConf);
        // hide spinner as default
        this._isVisible = false;
    }
    /**
     * @param {?} condition
     * @return {?}
     */
    set visible(condition) {
        if (condition)
            this.show();
        else
            this.hide();
    }
    /**
     * @return {?}
     */
    get visible() {
        return this._isVisible;
    }
    /**
     * @param {?} conf
     * @return {?}
     */
    set config(conf) {
        // override with spinner config
        this.assignConfig(conf);
    }
    /**
     * @return {?}
     */
    get config() {
        return this._config;
    }
    /**
     * @return {?}
     */
    show() {
        this._isVisible = true;
    }
    /**
     * @return {?}
     */
    hide() {
        this._isVisible = false;
    }
    /**
     * @return {?}
     */
    toggle() {
        this._isVisible = !this._isVisible;
    }
    /**
     * @param {?} conf
     * @return {?}
     */
    assignConfig(conf) {
        let /** @type {?} */ cnf = Object.assign(conf);
        for (let /** @type {?} */ prop in cnf) {
            if (cnf.hasOwnProperty(prop))
                this._config[prop] = cnf[prop];
        }
    }
}
NgxSpinner.decorators = [
    { type: Component, args: [{
                selector: 'ngx-spinner',
                template: `<!-- spinner -->
<ng-template *ngIf="visible" [spinner-animation]="config"></ng-template>`,
                styles: [`:host.block_window{align-items:center;justify-content:center;position:fixed;display:flex;width:100%;height:100%;top:0;left:0;z-index:10002}:host.block_ui{align-items:center;justify-content:center;position:absolute;display:flex;width:100%;height:100%;top:0;left:0;z-index:10001}:host.inplace{position:absolute;background:0 0!important;display:flex;width:100%;height:100%;top:0;left:0;align-items:center;justify-content:start}`],
                host: { '[class]': 'visible?config.placement:""', '[style.background]': 'visible?config.bgColor:""' }
            },] },
];
/** @nocollapse */
NgxSpinner.ctorParameters = () => [
    { type: NgxSpinnerService }
];
NgxSpinner.propDecorators = {
    visible: [{ type: Input }],
    config: [{ type: Input }]
};
function NgxSpinner_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxSpinner.prototype._config;
    /** @type {?} */
    NgxSpinner.prototype._isVisible;
    /** @type {?} */
    NgxSpinner.prototype.ngxSpinnerService;
}
